### ISSUE TITLE
In v2.0.3, the new canvas-prebuilt dependency causes compilation errors in Kibana

### ISSUE THREAD

**Issue by nyurik** [vega/vega-scenegraph#55]
The new `canvas-prebuilt` module added in v2.0.3 starts causing errors when using vega npm module.

```
    ERROR in .-vega-vis/~/canvas-prebuilt/canvas/lib/bindings.js
    Module not found: Error: Cannot resolve 'file' or 'directory' ../build/Release/canvas in /home/yurik/dev/elastic/kibana-vega-vis/node_modules/canvas-prebuilt/canvas/lib
     @ .-vega-vis/~/canvas-prebuilt/canvas/lib/bindings.js 3:17-51
```

**Comment by jheer** [comment]
Please include your OS, node version, etc to enable replication.

**Comment by nyurik** [comment]
@jheer sadly, I don't know how to replicate. My [Kibana module](https://github.com/nyurik/kibana-vega-vis) uses Vega, and is in turn used by [Kibana](https://github.com/elastic/kibana) itself.  Kibana is a nodejs service that automatically compiles modules with all of their dependencies, using rollup (?) and webpack.  The previous version of Vega worked fine with this setup, but now I see the above error from this stack. It seems the error happens deep inside the webpack, when it tries to combine all of the code including plugins with their dependencies.

```
    ERROR in .-vega-vis/~/canvas-prebuilt/canvas/lib/bindings.js
    Module not found: Error: Cannot resolve 'file' or 'directory' ../build/Release/canvas in /home/yurik/dev/elastic/kibana-vega-vis/node_modules/canvas-prebuilt/canvas/lib
     @ .-vega-vis/~/canvas-prebuilt/canvas/lib/bindings.js 3:17-51

    at Object.exports.create (/home/yurik/dev/elastic/kibana/node_modules/boom/lib/index.js:21:17)
    at LazyOptimizer.failedStatsToError (/home/yurik/dev/elastic/kibana/src/optimize/base_optimizer.js:224:17)
    at Compiler.<anonymous> (/home/yurik/dev/elastic/kibana/src/optimize/lazy/lazy_optimizer.js:42:24)
    at Compiler.applyPlugins (/home/yurik/dev/elastic/kibana/node_modules/tapable/lib/Tapable.js:26:37)
    at Watching._done (/home/yurik/dev/elastic/kibana/node_modules/webpack/lib/Compiler.js:78:17)
    at Watching.<anonymous> (/home/yurik/dev/elastic/kibana/node_modules/webpack/lib/Compiler.js:51:17)
    at /home/yurik/dev/elastic/kibana/node_modules/webpack/lib/Compiler.js:403:12
    at Compiler.next (/home/yurik/dev/elastic/kibana/node_modules/tapable/lib/Tapable.js:67:11)
    at Compiler.<anonymous> (/home/yurik/dev/elastic/kibana/node_modules/webpack/lib/CachePlugin.js:40:4)
    at Compiler.applyPluginsAsync (/home/yurik/dev/elastic/kibana/node_modules/tapable/lib/Tapable.js:71:13)
    at Compiler.<anonymous> (/home/yurik/dev/elastic/kibana/node_modules/webpack/lib/Compiler.js:400:9)
    at Compilation.<anonymous> (/home/yurik/dev/elastic/kibana/node_modules/webpack/lib/Compilation.js:577:13)
    at Compilation.applyPluginsAsync (/home/yurik/dev/elastic/kibana/node_modules/tapable/lib/Tapable.js:60:69)
    at Compilation.<anonymous> (/home/yurik/dev/elastic/kibana/node_modules/webpack/lib/Compilation.js:572:10)
    at Compilation.applyPluginsAsync (/home/yurik/dev/elastic/kibana/node_modules/tapable/lib/Tapable.js:60:69)
    at Compilation.<anonymous> (/home/yurik/dev/elastic/kibana/node_modules/webpack/lib/Compilation.js:567:9)
```
I'm running on [Kibana 5.5](https://github.com/elastic/kibana/tree/5.5)   80def1071

My nodejs is 6.11.3 on Ubuntu x64 17.04

**Comment by nyurik** [comment]
P.S. You could try to replicate by
* $ mkdir elastic && cd elastic
* Download https://www.elastic.co/downloads/elasticsearch v5.6, unzip to elasticsearch, and run it with bin/elasticsearch
* $ cd .. && git clone https://github.com/elastic/kibana.git && cd kibana
* $ git checkout 5.6 && npm install
* $ cd .. && git clone https://github.com/nyurik/kibana-vega-vis.git && cd kibana-vega-vis && npm install
* $ npm start

This way you have two processes running - elasticsearch, and Kibana.  Kibana is ran by plugin in the debug mode.  Use http://localhost:5601, and Kibana will try to compile UI, and fail with the above

**Comment by abudaan** [comment]
Same issue with Browserify. To reproduce:

* `git clone git@github.com:abudaan/vega3.03-canvas-prebuilt-issue`
* `yarn install` / `npm install`
* `npm run bundle`

**Comment by abudaan** [comment]
My guess is that it is an issue in canvas-prebuilt and not in vega. If I bundle a file that only contains a canvas import statement `import canvas from 'canvas'` I get this message:

SyntaxError: Unexpected character '' (2:0) while parsing [...] /node_modules/canvas/build/Release/canvas.node

**Comment by nyurik** [comment]
I think there should be a Vega release without canvas-prebuilt for now, until this is resolved, because the latest release fixes some important issues.  My understanding is that canvas-prebuilt is not needed for any browser-based operations anyway.

**Comment by jheer** [comment]
Thanks @nyurik for reporting the issue and thanks @abudaan for the easy replication in browserify! 

The problem was that Vega's package.json was missing `"browser": {"canvas-prebuilt": false}` to ensure exclusion of the package when bundling for the web. (The vega-scenegraph and vega-wordcloud modules were properly configured, but this detail was erroneously left out of the main vega repo.) I've fixed the issue and released v3.0.4.

I can confirm that browserify now bundles without any issues. @nyurik please confirm if Vega 3.0.4 fixes the issue for you in Kibana as well. I'm closing the issue for now, but feel free to re-open if you determine that problems remain.
